packages/rx_bloc_cli/mason_templates/bricks/feature_onboarding/__brick__/lib/feature_onboarding_email_confirmation/blocs/onboarding_email_confirmed_bloc.dart

```diff
@@ -0,0 +1,76 @@
+{{> licence.dart }}
+
+import 'package:rx_bloc/rx_bloc.dart';
+import 'package:rxdart/rxdart.dart';
+
+import '../../base/common_services/users_service.dart';
+import '../../base/extensions/error_model_extensions.dart';
+import '../../base/models/errors/error_model.dart';
+import '../../lib_router/blocs/router_bloc.dart';
+import '../../lib_router/router.dart';
+
+part 'onboarding_email_confirmed_bloc.rxb.g.dart';
+
+/// A contract class containing all events of the OnboardingEmailConfirmedBloC.
+abstract class OnboardingEmailConfirmedBlocEvents {
+  void fetchData();
+
+  void goToLogin();
+
+  void goToPhonePage();
+}
+
+/// A contract class containing all states of the OnboardingEmailConfirmedBloC.
+abstract class OnboardingEmailConfirmedBlocStates {
+  /// The loading state
+  Stream<bool> get isLoading;
+
+  /// The error state
+  Stream<ErrorModel> get errors;
+
+  ConnectableStream<bool> get data;
+
+  ConnectableStream<void> get onRouting;
+}
+
+@RxBloc()
+class OnboardingEmailConfirmedBloc extends $OnboardingEmailConfirmedBloc {
+  OnboardingEmailConfirmedBloc(
+    this._verifyEmailToken,
+    this._usersService,
+    this._routerBloc,
+  ) {
+    onRouting.connect().addTo(_compositeSubscription);
+    data.connect().addTo(_compositeSubscription);
+  }
+
+  final String _verifyEmailToken;
+  final UsersService _usersService;
+  final RouterBlocType _routerBloc;
+
+  @override
+  ConnectableStream<bool> _mapToDataState() => _$fetchDataEvent
+      .startWith(null)
+      .throttleTime(const Duration(milliseconds: 200))
+      .switchMap((value) => _usersService
+          .confirmEmail(token: _verifyEmailToken.split('---')[0])
```

Comment:
Could you provide some context on this line of code?

Reply:
It was an implementation detail from a... certain project that I forgot to remove. Good catch, I've removed it.
I believe we shouldn't do any token parsing here, since it will be decided on a per-project basis